ConfigDescriptionI18nTest unstable

This test failed in the Travis CI Java 11 build:

```
TEST assertConfigDescriptionsAreLocalized(org.eclipse.smarthome.config.xml.test.ConfigDescriptionI18nTest) <<< ERROR: 
Expected: is <2>
     but: was <3>
java.lang.AssertionError: 
Expected: is <2>
     but: was <3>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.eclipse.smarthome.config.xml.test.BindingInstaller.lambda$0(BindingInstaller.java:50)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:152)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:120)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:73)
	at org.eclipse.smarthome.config.xml.test.BindingInstaller.exec(BindingInstaller.java:50)
	at org.eclipse.smarthome.config.xml.test.ConfigDescriptionI18nTest.assertConfigDescriptionsAreLocalized(ConfigDescriptionI18nTest.java:53)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at aQute.junit.Activator.test(Activator.java:326)
	at aQute.junit.Activator.automatic(Activator.java:220)
	at aQute.junit.Activator.run(Activator.java:176)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at aQute.launcher.Launcher.launch(Launcher.java:385)
	at aQute.launcher.Launcher.run(Launcher.java:143)
	at aQute.launcher.Launcher.main(Launcher.java:121)
	at aQute.launcher.pre.EmbeddedLauncher.executeWithRunPath(EmbeddedLauncher.java:154)
	at aQute.launcher.pre.EmbeddedLauncher.findAndExecute(EmbeddedLauncher.java:119)
	at aQute.launcher.pre.EmbeddedLauncher.main(EmbeddedLauncher.java:51)

```

--- a/itests/org.openhab.core.config.xml.tests/src/main/java/org/openhab/core/config/xml/test/ConfigDescriptionI18nTest.java
+++ b/itests/org.openhab.core.config.xml.tests/src/main/java/org/openhab/core/config/xml/test/ConfigDescriptionI18nTest.java
@@ -20,13 +20,15 @@ import java.util.LinkedList;
 import java.util.Locale;
 import java.util.stream.Collectors;
 
+import org.junit.Before;
+import org.junit.Test;
 import org.openhab.core.config.core.ConfigDescription;
 import org.openhab.core.config.core.ConfigDescriptionParameter;
 import org.openhab.core.config.core.ConfigDescriptionParameterGroup;
-import org.openhab.core.config.core.ConfigDescriptionRegistry;
+import org.openhab.core.config.core.ConfigDescriptionProvider;
+import org.openhab.core.test.BundleCloseable;
+import org.openhab.core.test.SyntheticBundleInstaller;
 import org.openhab.core.test.java.JavaOSGiTest;
-import org.junit.Before;
-import org.junit.Test;
 
 /**
  * The ConfigDescriptionI18nTest is a test for loading of configuration description from XML documents.
@@ -38,38 +40,40 @@ public class ConfigDescriptionI18nTest extends JavaOSGiTest {
 
     private static final String TEST_BUNDLE_NAME = "yahooweather.bundle";
 
-    private ConfigDescriptionRegistry configDescriptionRegistry;
-    private BindingInstaller bindingInstaller;
+    private ConfigDescriptionProvider configDescriptionProvider;
 
     @Before
     public void setUp() {
-        configDescriptionRegistry = getService(ConfigDescriptionRegistry.class);
-        assertThat(configDescriptionRegistry, is(notNullValue()));
-        bindingInstaller = new BindingInstaller(this::waitForAssert, configDescriptionRegistry, bundleContext);
+        configDescriptionProvider = getService(ConfigDescriptionProvider.class,
+                serviceReference -> "core.xml.config".equals(serviceReference.getProperty("esh.scope")));
+        assertThat(configDescriptionProvider, is(notNullValue()));
     }
 
     @Test
     public void assertConfigDescriptionsAreLocalized() throws Exception {
-        bindingInstaller.exec(TEST_BUNDLE_NAME, () -> {
-            Collection<ConfigDescription> configDescriptions = configDescriptionRegistry
+        try (BundleCloseable bundle = new BundleCloseable(
+                SyntheticBundleInstaller.install(bundleContext, TEST_BUNDLE_NAME))) {
+            assertThat(bundle, is(notNullValue()));
+
+            Collection<ConfigDescription> configDescriptions = configDescriptionProvider
                     .getConfigDescriptions(Locale.GERMAN);
 
             ConfigDescription config = new LinkedList<>(configDescriptions).getFirst();
             assertThat(config, is(notNullValue()));
 
             String expected = "location.label = Ort\n" + //
-            "location.description = Ort der Wetterinformation.\n" + //
-            "unit.label = Einheit\n" + //
-            "unit.description = Spezifiziert die Einheit der Daten. Valide Werte sind 'us' und 'metric'\n" + //
-            "refresh.label = Aktualisierungsintervall\n" + //
-            "refresh.description = Spezifiziert das Aktualisierungsintervall in Sekunden\n" + //
-            "question.pattern = Wie ist das Wetter in [\\w]*?\n" + //
-            "question.options = München, Köln\n" + //
-            "group.label = Group 1 German Label\n" + //
-            "group.description = Group 1 German Description";
+                    "location.description = Ort der Wetterinformation.\n" + //
+                    "unit.label = Einheit\n" + //
+                    "unit.description = Spezifiziert die Einheit der Daten. Valide Werte sind 'us' und 'metric'\n" + //
+                    "refresh.label = Aktualisierungsintervall\n" + //
+                    "refresh.description = Spezifiziert das Aktualisierungsintervall in Sekunden\n" + //
+                    "question.pattern = Wie ist das Wetter in [\\w]*?\n" + //
+                    "question.options = München, Köln\n" + //
+                    "group.label = Group 1 German Label\n" + //
+                    "group.description = Group 1 German Description";
 
             assertEquals(expected, asString(config));
-        });
+        }
     }
 
     private static String asString(ConfigDescription description) {

--- a/itests/org.openhab.core.config.xml.tests/src/main/java/org/openhab/core/config/xml/test/ConfigDescriptionI18nTest.java
+++ b/itests/org.openhab.core.config.xml.tests/src/main/java/org/openhab/core/config/xml/test/ConfigDescriptionI18nTest.java
@@ -15,8 +15,9 @@ package org.openhab.core.config.xml.test;
 import static org.hamcrest.CoreMatchers.*;
 import static org.junit.Assert.*;
 
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.Collection;
-import java.util.LinkedList;
 import java.util.Locale;
 import java.util.stream.Collectors;
 
@@ -58,7 +59,7 @@ public class ConfigDescriptionI18nTest extends JavaOSGiTest {
             Collection<ConfigDescription> configDescriptions = configDescriptionProvider
                     .getConfigDescriptions(Locale.GERMAN);
 
-            ConfigDescription config = new LinkedList<>(configDescriptions).getFirst();
+            ConfigDescription config = findDescription(configDescriptions, "config:Dummy");
             assertThat(config, is(notNullValue()));
 
             String expected = "location.label = Ort\n" + //
@@ -99,6 +100,18 @@ public class ConfigDescriptionI18nTest extends JavaOSGiTest {
         return sb.toString();
     }
 
+    private static ConfigDescription findDescription(Collection<ConfigDescription> descriptions, String uri) {
+        try {
+            return findDescription(descriptions, new URI(uri));
+        } catch (URISyntaxException e) {
+            return null;
+        }
+    }
+
+    private static ConfigDescription findDescription(Collection<ConfigDescription> descriptions, URI uri) {
+        return descriptions.stream().filter(d -> uri.equals(d.getUID())).findFirst().get();
+    }
+
     private static ConfigDescriptionParameter findParameter(ConfigDescription description, String parameterName) {
         return description.getParameters().stream().filter(p -> parameterName.equals(p.getName())).findFirst().get();
     }
@@ -108,5 +121,4 @@ public class ConfigDescriptionI18nTest extends JavaOSGiTest {
         return description.getParameterGroups().stream().filter(g -> parameterGroupName.equals(g.getName())).findFirst()
                 .get();
     }
-
 }

